CustomResourceDefinition/v1alpha2/bool/main.tf

```diff
@@ -0,0 +1,28 @@
+
+locals {
+  declared_default_field = contains(keys(var.manifest), "default")
+}
+
+output "schema" {
+  value = {
+    type    = "bool"
+    version = "v1"
+    subItem = null
+    validations = {
+      has_default_value = can(var.manifest.default) ? true : false
+      default_value     = try(var.manifest.default, null)
+    }
+  }
+
+  precondition {
+    condition     = !(
+      local.declared_default_field && try(var.manifest.default, null) == null
+  )
```

Comment:
Aqui não deveria ser `tem declaração do default e o default for diferente de null ou não tem declaração do default` (`condition = (local.declared_default_field && try(var.manifest.default, null) != null) || !local.declared_default_field` ?